go-controller/pkg/util/net.go

```diff
@@ -30,37 +30,32 @@ func intToIP(i *big.Int) net.IP {
 }
 
 // GetPortAddresses returns the MAC and IPs of the given logical switch port
-func GetPortAddresses(portName string) (net.HardwareAddr, []net.IP, error) {
-	out, stderr, err := RunOVNNbctl("--if-exists", "get", "logical_switch_port", portName, "dynamic_addresses", "addresses")
-	if err != nil {
-		return nil, nil, fmt.Errorf("error while obtaining dynamic addresses for %s: stdout: %q, stderr: %q, error: %v",
-			portName, out, stderr, err)
+func GetPortAddresses(portName string, ovnNBClient goovn.Client) (net.HardwareAddr, []net.IP, error) {
+	lsp, err := ovnNBClient.LSPGet(portName)
+	if err != nil || lsp == nil {
+		// --if-exists handling in goovn
+		if err == goovn.ErrorSchema || err == goovn.ErrorNotFound {
+			return nil, nil, nil
+		}
+		return nil, nil, err
 	}
 
-	if out == "" {
-		klog.V(5).Infof("Port %s doesn't exist in the nbdb while fetching"+
-			" port addresses", portName)
-		return nil, nil, nil
-	}
-	// Convert \r\n to \n to support Windows line endings
-	out = strings.Replace(out, "\r\n", "\n", -1)
-	addresses := strings.Split(out, "\n")
-	out = addresses[0]
-	if out == "[]" {
-		out = addresses[1]
+	var addresses []string
+
+	if lsp.DynamicAddresses == "" {
```

Comment:
with the changes for disabling OVN assigning dynamic ip addresses and instead using ovn-k8s ip allocator, why do we even support dynamic addresses in the code anymore?

Reply:
I have left the code as is because the `GetPortAddresses` API is called from a couple of other places as well. Also, in the unlikely event that prior to an upgrade from a build that predates IPAM changes to the ones where IPAM is done in ovn-k-master, there were some LSPs that had northd allocating IP addresses, but addLogicalPort hadn't written the annotation to the API server, we may have to handle the corner case.